test/integration/developer_portal/signup_test.rb

```diff
@@ -26,6 +26,48 @@ def test_create
     assert_response :redirect
   end
 
+  # A post request can't be sent directly to the controller bypassing the form
+  # when the spam protection is always enabled.
+  def test_create_spam_protected_always
+    @provider.settings.update(spam_protection_level: :captcha)
+    DeveloperPortal::SignupController.any_instance.stubs(:verify_captcha).returns(false)
+
+    post signup_path(account: {
+                       promo_code: '12345',
+                       org_name:   'alaska',
+                       user: {
+                         email:    'foo@example.edu',
+                         username: 'supertramp',
+                         password: 'westisthebest'
+                       }
+                     })
+
+    # The user wasn't created, so the spam protection worked
+    user = @provider.buyer_users.find_by(email: 'foo@example.edu')
+    assert_nil user
```

Comment:
Comments are usually a sign of improvable code. Instead of adding this comment, try to re-write in a more readable way. I don't think neither is required here though, but maybe the test title could be more descriptive. Also consider adding test classes to group different conditions, such as `SignupTestWithCatpcha`, `SignupTestWithoutCatpcha`, `SignupTestWithAutoCatpcha`. This way test cases can be titled as:
```ruby
class SignupTestWithCatpcha
  test 'user should not be created if captcha fails' do
    ...
  end

  test 'user should be created if captcha passes' do
    ...
  end
end
```
Therefore reducing the need for comments.

Also a small suggestion for the assertion, query `Buyer` instead of the provider to make the test stricter?
```suggestion
    assert_nil Buyer.find_by(email: test_email)
```

Reply:
`Buyer` is not a class, I used `User` instead. Why is `Buyer` more strict than `provider.buyer_users`? Is it because it checks that the user wasn't created at any place, not only the provider? In that case `User` is even more strict, right?

On the other hand, I tried to place the tests inside a subclass, but then I realized than all tests must be part of a subclass, otherwise the tests that doesn't belong to a subclass are executed many times, due to inheritance. So I refactored the whole suite into subclasses: one subclass per each action and inside the `Create` action subclass, one subclass for each captcha state.

Reply:
Forget what I said